### PR TITLE
RMMIS-6456: Fieldset ES6ify, update tests, remove state

### PR DESCRIPTION
**Pull request by traviscooperscdhhs**

### CODE REVIEW

File: src/Fieldset.jsx [hunk 1]
```diff
@@ -1,64 +1,71 @@
...
     if(this.props.helpText){
       helpText = <p key="help-block">{this.props.helpText}</p>;
     }
     return helpText;
-  },
+  }
 
-  getDescription: function(){
-    var description;
+  getDescription(){
```
Comment (kjschulz) [inline src/Fieldset.jsx]: I think we are getting rid of the `Description` component, right?  I would recommend doing something like what @michaelleeallen did in the `Field` component.  Like this:
<code>
  renderDescription() {
    if (this.props.description) {
      let popover = <Popover title={this.props.descriptionTitle}>{this.props.description}</Popover>;
      return (
        <span className="field-description">
          <OverlayTrigger
            trigger={this.descriptionTrigger}
            placement={this.descriptionPlacement}
            overlay={popover}>
            <Glyphicon glyph="info-sign" aria-hidden="true"/>
          </OverlayTrigger>
        </span>
      );
    }
  }
</code>

File: src/Fieldset.jsx [hunk 2]
```diff
@@ -1,64 +1,71 @@
...
-      componentType: 'fieldset'
-    };
-  },
+  constructor(props) {
+    super(props);
+  }
 
-  getHelpText: function(){
-    var helpText;
+  getHelpText(){
```
Comment (kjschulz) [inline src/Fieldset.jsx]: Kinda nitpickey I know, but just a suggestion to be a little simpler:
<code>
  renderHelpText() {
    if (this.props.helpText) {
      return <p key="help-block">{this.props.helpText}</p>;
    }
  }
</code>

File: src/Fieldset.jsx [hunk 3]
```diff
@@ -1,64 +1,71 @@
...
+  getDescription(){
+    let description;
     if(this.props.description){
       description = <Description key="description-text" {...this.props} />;
     }
     return description;
-  },
+  }
 
-  getLabel: function(){
```
Comment (kjschulz) [inline src/Fieldset.jsx]: Same thing here and also should probably use ES6 template strings for the key:
<code>
  renderLabel() {
    if (this.props.name) {
      return (
        <legend className="field-label" key={`${this.props.name}legend`}>
          {this.props.legend} {this.renderDescription()}
        </legend>
      );
    }
  }
</code>

Comment (kjschulz) [inline src/Fieldset.jsx]: Other than that, lookin' good Holmes!

**Comment by kjschulz** [comment]
:+1:

**Comment by mallenscdhhs** [comment]
:+1:

**Pull request merged**